persistence/subsystem/offsetstore/transaction.go

```diff
@@ -0,0 +1,25 @@
+package offsetstore
+
+import (
+	"context"
+	"errors"
+)
+
+// ErrConflict is returned when the offset supplied as a parameter to method
+// Transaction.SaveOffset does not equal to the currently persisted offset.
+var ErrConflict = errors.New("optimistic concurrency conflict in the application event stream offset")
```

Comment:
```suggestion
var ErrConflict = errors.New("an optimistic concurrency conflict occured while persisting to the offset store")
```

I will change `ErrConflict` in `queuestore` to match this kind of phrasing.